src/newgrf.cpp

```diff
@@ -7150,9 +7148,10 @@ static void GRFLoadError(ByteReader *buf)
 	}
 
 	/* For now we can only show one message per newgrf file. */
-	if (_cur.grfconfig->error != nullptr) return;
+	if (_cur.grfconfig->error.get() != nullptr) return;
```

Comment:
```suggestion
	if (_cur.grfconfig->error) return;
```
`operator bool` of a `unique_ptr` checks if it has a value assigned.

Reply:
We've traditionally deliberately avoided nullptr-is-false tests, I know it's not exactly the same but the appearance is similar. Do we want to revise that?

In CODINGSTYLE.md: `* Don't treat non-flags as flags: use "if (char_pointer != nullptr && *char_pointer != '\0')", not "if (char_pointer && *char_pointer)".`

Pretty sure you can also do `if (_cur.grfconfig->error != nullptr) return;` without the get() to stick with current coding style.

Reply:
I wasn't aware that they special-cases the `==` and `!=` operators to allow `nullptr`. Changed it to just compare to `nullptr`.